src/OptimisticToggle.js

```diff
@@ -1,22 +1,36 @@
+// @flow
 import React, { Component } from 'react';
 
 const noop = () => {};
 
-class OptimisticToggle extends Component {
-  static defaultProps = {
+type Props = {
+  /** Initial Value of the Checkbox */
+  initialValue?: boolean,
+  /** On-change handler that returns a Promise Object */
+  action?: (toggleState: boolean, event: React.SyntheticEvent) => Promise<any>,
```

Comment:
can we do named import of Synthetic import or destructure it. React.SyntheticEvent makes that line much bigger and thus harder to read.